contracts/toCKB-typescript/src/switch/mod.rs

```diff
@@ -132,6 +136,32 @@ fn get_deletion_tx_type(data: &ToCKBCellDataView) -> Result<TxType, Error> {
     }
 }
 
+fn verify_xt(tx_type: &TxType) -> Result<(), Error> {
+    use TxType::*;
+    match tx_type {
+        MintXT
+        | PretermRedeem
+        | AttermRedeem
+        | AuctionSignerTimeout
+        | AuctionUnderCollateral
+        | AuctionFaultyWhenWarranty
+        | AuctionFaultyWhenRedeeming => Ok(()),
+        _ => forbid_mint_xt(),
+    }
+}
+
+fn forbid_mint_xt() -> Result<(), Error> {
+    let sudt_cell_count = QueryIter::new(load_cell_type, Source::Output)
+        .filter(|type_opt| type_opt.is_some())
+        .map(|type_opt| type_opt.unwrap())
+        .filter(|script| script.code_hash().raw_data().as_ref() == SUDT_CODE_HASH.as_ref())
+        .count();
+    if 0 != sudt_cell_count {
+        return Err(Error::TxInvalid);
```

Comment:
It is better to add an error type to tell user 'forbid_mint_xt'

Reply:
don't want to explode error, only malicious user  would do that